packages/realm/src/app-services/App.ts

```diff
@@ -277,6 +286,10 @@ export class App<
       }
     }
 
+    if (fetchOverride) {
+      inject({ fetch: fetchOverride } as NetworkType);
+    }
```

Comment:
If we decide to add a more explicit type than `any` or `unknown` to the `fetchOverride` in the app config (I think at least specifying that it should be a function could be good), then what do you think of adding an explicit check for `undefined` here? This way, we can let users know if they have provided an invalid falsy value *in addition to* an invalid truthy value. Also, we can assert that it's a function.
```suggestion
    if (fetchOverride !== undefined) {
      assert.function(fetchOverride, "fetchOverride");
      inject({ fetch: fetchOverride } as NetworkType);
    }
```